proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -2020,33 +2034,80 @@ lemma set_priority_valid_sched[wp]:
   apply (auto simp: valid_sched_def valid_sched_action_def st_tcb_at_def obj_at_def not_cur_thread_def split: thread_state.splits)
   done
 
+lemma set_object_tcb_etcb_at:
+  "\<lbrace> \<lambda>s. Q (etcb_at P p s) \<and> (p = r \<longrightarrow> Q (P (etcb_of tcb))) \<rbrace>
+   set_object r (TCB tcb)
+   \<lbrace> \<lambda>_ s. Q (etcb_at P p s) \<rbrace>"
+  by (auto simp: valid_def in_monad etcb_at_def etcbs_of'_def etcb_of_def set_object_def get_object_def
+              split: kernel_object.splits)
+
+lemma thread_set_no_change_etcb_at:
+  assumes x: "\<And>tcb. Q (P (etcb_of (f tcb))) = Q (P (etcb_of tcb))"
+  shows      "thread_set f t' \<lbrace>\<lambda>s. Q (etcb_at P t s)\<rbrace>"
+  apply (simp add: thread_set_def pred_tcb_at_def)
+  apply wp
+   apply (rule set_object_tcb_etcb_at)
+  apply wp
+  apply (clarsimp simp: etcb_at_def)
+  apply (drule get_tcb_SomeD)
+  apply (clarsimp simp: x etcbs_of'_def)
+  done
```

Comment:
this proof looks a bit old (assumes x, wp+simp vs wpsimp), is it moved?

Reply:
I think it started as a copy of a old proof, possibly `thread_set_no_change_tcb_pred`.

Reply:
Maybe worth a cleanup then?